source/class/qx/ui/form/validation/Manager.js

```diff
@@ -550,12 +560,9 @@ qx.Class.define("qx.ui.form.validation.Manager", {
       // add the forms fail message
       if (!this.isValid()) {
         let msg = this.getInvalidMessage();
-        if (!msg && qx.core.Environment.get("qx.ui.form.validation.Manager.allowDefaultInvalidMessage")) {
-          msg = "Invalid field";
-        } else if (qx.core.Environment.get("qx.debug")) {
-          this.assertTrue(msg !== null && msg.length > 0);
```

Comment:
Why is this removed?  It should remain, although with the locale changes for `msg = qx.locale.Manager.tr("Invalid field");`

Reply:
You did not change the invalidmessage handling of the form. So the default is '' as before.
This check you always get an invalid field error even if there is no other error.

Reply:
the previous code was to assume a default message if there is none but when the form is still invalid.  By removing this code, then invalid forms will not have a message even if they are invalid - unless they have widgets which are also invalid.  AIUI the code should go back in

Reply:
Form validation works differnt from formItem validation.
The invalidMessage is not the message you send to the user it's the set from exception message, see:

Manager.js, line 354ff

```
// reset the invalidMessage
      this.setInvalidMessage("");

      if (this.__isAsyncValidator(formValidator)) {
        this.__asyncResults[this.toHashCode()] = null;
        formValidator.validateForm(items, this, context);
        return null;
      }

      try {
        var formValid = formValidator.call(context, items, this);
        if (formValid === undefined) {
          formValid = true;
        }
      } catch (e) {
        if (e instanceof qx.core.ValidationError) {
          formValid = false;

          if (e.message && e.message != qx.type.BaseError.DEFAULTMESSAGE) {
            var invalidMessage = e.message;
          } else {
            var invalidMessage = e.getComment();
          }
          this.setInvalidMessage(invalidMessage);
        } else {
          throw e;
        }
      }
      return formValid;
    },
```

Reply:
With this the test of empty invalidMessage of the form fails for ever,

Reply:
Same name for different functionality - should we change this in v8?